lib/edoc/src/edoc_layout_chunks.erl

```diff
@@ -0,0 +1,512 @@
+    format_element(rewrite_a_tag(E));
+format_content_(#xmlElement{} = E) ->
+    format_element(E).
+
+format_element(#xmlElement{} = E) ->
+    #xmlElement{name = Name, content = Content, attributes = Attributes} = E,
+    case {is_edoc_tag(Name), is_html_tag(Name)} of
+	{true, _} ->
+	    format_content(Content);
+	{_, false} ->
+	    edoc_report:warning("'~s' is not accepted - skipping tag, extracting content", [Name]),
+	    format_content(Content);
+	_ ->
+	    [{Name, format_attributes(Attributes), format_content(Content)}]
+    end.
+
+-spec format_attributes([xmerl_attribute()]) -> [shell_docs:chunk_element_attr()].
+format_attributes(Attrs) ->
+    [ format_attribute(Attr) || Attr <- Attrs ].
+
+-spec format_attribute(xmerl_attribute()) -> shell_docs:chunk_element_attr().
+format_attribute(#xmlAttribute{} = Attr) ->
+    #xmlAttribute{name = Name, value = V} = Attr,
+    %% From xmerl.hrl: #xmlAttribute.value :: IOlist() | atom() | integer()
+    case V of
+	_ when is_list(V)    -> {Name, unicode:characters_to_binary(V)};
+	_ when is_atom(V)    -> {Name, atom_to_binary(V, utf8)};
+	_ when is_integer(V) -> {Name, integer_to_binary(V)}
+    end.
+
+-spec is_edoc_tag(atom()) -> boolean().
+is_edoc_tag(fullDescription) -> true;
+is_edoc_tag(since) -> true;
+is_edoc_tag(_) -> false.
+
+-spec is_html_tag(atom()) -> boolean().
+is_html_tag(Tag) ->
+    %% This is only a subset of existing HTML tags.
+    %% Compare with https://developer.mozilla.org/en-US/docs/Web/HTML/Element
+    Tags = [a,p,h1,h2,h3,i,br,em,pre,code,ul,ol,li,dl,dt,dd],
+    lists:member(Tag, Tags).
+
+rewrite_a_tag(#xmlElement{name = a} = E) ->
+    SimpleE = xmerl_lib:simplify_element(E),
+    xmerl_lib:normalize_element(rewrite_docgen_link(SimpleE)).
+
+rewrite_see_tags([]) -> [];
+rewrite_see_tags([#xmlElement{name = see} | _] = SeeTags) ->
+    Grouped = [ rewrite_see_tag(T) || T <- SeeTags ],
+    NewXML = {p, [], ["See also "] ++ lists:join(", ", Grouped) ++ ["."]},
+    %% Convert strings to binaries in the entire new tree:
+    [format_content_(xmerl_lib:normalize_element(NewXML))].
+
+rewrite_see_tag(#xmlElement{name = see} = E) ->
+    %% TODO: this is not formatted nicely by shell_docs...
+    %% missing `p' around preceding description
+    SeeTag = xmerl_lib:simplify_element(E),
+    {see, Attrs, XML} = rewrite_docgen_link(SeeTag),
+    {a, Attrs, XML}.
+
+rewrite_docgen_link({Tag, AttrL, SubEls} = E) when Tag =:= a; Tag =:= see ->
+    Attrs = maps:from_list(AttrL),
+    case {maps:get('docgen-rel', Attrs, false), maps:get('docgen-href', Attrs, false)} of
+	{false, false} -> E;
+	{false, _} -> inconsistent_docgen_attrs(Attrs);
+	{_, false} -> inconsistent_docgen_attrs(Attrs);
+	{ShortRel, URI} ->
+	    AttrsNoDocgen = maps:without(['docgen-rel', 'docgen-href'], Attrs),
+	    NewAttrs = AttrsNoDocgen#{rel => expand_docgen_rel(ShortRel),
+				      href => URI},
+	    {Tag, [{'edoc-orig-tag', Tag}] ++ maps:to_list(NewAttrs), SubEls}
```

Comment:
> What is the purpose of the 'edoc-orig-tag'?

EDoc accepts any HTML, so also `<a>` tags. Putting `edoc-orig-tag` in `<a>`s which are generated makes it easy to debug / test. I'm not sure if it could be of any use to the final consumers - I'll clean it up.

Reply:
> The name attribute seems misplaced here...

Good catch, though this comes from EDoc core, not my changes. I'm removing it with the hope that nothing breaks 🤞

Reply:
> Good catch, though this comes from EDoc core, not my changes. I'm removing it with the hope that nothing breaks :crossed_fingers:

Then the HTML backend removes it somehow as well, as I don't think it ends up in the final HTML.